src/ui-test/server/ui/rspServerProvider.ts

```diff
@@ -0,0 +1,77 @@
+import { AdaptersConstants } from "../../common/adaptersContants";
+import { ServerState } from "../../common/serverState";
+import { serverHasState } from "../serverUtils";
+
+
+/**
+ * RSP Server Provider item representation
+ * @author Ondrej Dockal <odockal@redhat.com>
+ */
+export class RSPServerProvider {
+
+    private _serverProviderItem: ViewItem;
+
+    constructor(item: ViewItem) {
+        this._serverProviderItem = item;
+    }
+
+    public get serverProviderItem(): ViewItem {
+        if (this._serverProviderItem && this._serverProviderItem.isDisplayed()) {
+            return this._serverProviderItem;
+        } else {
+            throw Error('Server provider item (ViewItem object) is disposed or unavailable');
+        }
+    }
+
+    async performServerOperation(contextMenuItem: string, expectedState: ServerState, timeout: number) {
+        await this.serverProviderItem.select();
+        const menu = await this.serverProviderItem.openContextMenu();
+        const startAction = await menu.getItem(contextMenuItem);
+        await startAction.click();
+        await VSBrowser.instance.driver.wait(() => { return serverHasState(this, expectedState);}, timeout );
+    }
+
+    async startServerProvider(timeout: number=7000) {
+        await this.performServerOperation(AdaptersConstants.RSP_SERVER_PROVIDER_START, ServerState.Started, 10000);
+    }
+
+    async stopServerProvider(timeout: number = 7000) {
+        await this.performServerOperation(AdaptersConstants.RSP_SERVER_PROVIDER_STOP, ServerState.Stopped, 10000);
+    }
+
+    async terminateServerProvider(timeout: number = 7000) {
+        await this.performServerOperation(AdaptersConstants.RSP_SERVER_PROVIDER_TERMINATE, ServerState.Stopped, 10000);
+    }
+
+    async getCreateNewServerBox(): Promise<InputBox> {
+        const item = await (await this.serverProviderItem.openContextMenu()).getItem(AdaptersConstants.RSP_SERVER_PROVIDER_CREATE_NEW_SERVER);
+        await item.click();
+        return await InputBox.create();
+    }
+
+    async createNewServerCommand(): Promise<void> {
+        const input = await new Workbench().openCommandPrompt() as InputBox;
+        await input.setText('>' + AdaptersConstants.RSP_COMMAND + ' ' + AdaptersConstants.RSP_SERVER_PROVIDER_CREATE_NEW_SERVER);
+        await input.confirm();
+    }
+
+    async getNameLabel(): Promise<String> {
+        return this.serverProviderItem.getText();
+    }
+
+    async getServerName(): Promise<String> {
+        const text = await this.serverProviderItem.getText();
+        return text.slice(0, text.indexOf('(') - 1); 
+    }
+
+    async getServerStateLabel(): Promise<String> {
+        const text = await this.serverProviderItem.getText();
+        return text.slice(text.indexOf('(') + 1, text.indexOf(')'));
+    }
+
+    async getServerState(): Promise<ServerState> {
+        var label = (await this.getServerStateLabel()).toString();
```

Comment:
`await this.getServerStateLabel()` is a string already

Reply:
Type does not think so, without toString() next line show comp. comply:
```
Type 'String' cannot be used as an index type.ts(2538)
```

It seems to take it as String instead of string. Honestly, I do not understand this much. Java/Type-script stuff...

Reply:
I see, the problem is `String` vs `string` type mismatch. No idea why string literals are of a different type to string objects, but I recommend returning `string` in your methods.